fe/src/test/java/org/apache/doris/utframe/DemoTest.java

```diff
@@ -0,0 +1,193 @@
+import com.google.common.collect.Maps;
+
+import org.junit.Assert;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+import java.io.IOException;
+import java.util.List;
+import java.util.Map;
+import java.util.Random;
+
+/*
+ * This demo shows how to run unit test with mocked FE and BE.
+ * It will
+ *  1. start a mocked FE and a mocked BE.
+ *  2. Create a database and a tbl.
+ *  3. Make a schema change to tbl.
+ *  4. send a query and get query plan
+ */
+public class DemoTest {
+
+    private static int fe_http_port;
+    private static int fe_rpc_port;
+    private static int fe_query_port;
+    private static int fe_edit_log_port;
+
+    private static int be_heartbeat_port;
+    private static int be_thrift_port;
+    private static int be_brpc_port;
+    private static int be_http_port;
+    // use a unique dir so that it won't be conflict with other unit test which
+    // may also start a Mocked Frontend
+    private static String runningDir = "fe/mocked/DemoTest";
+
+    @BeforeClass
+    public static void beforeClass() throws EnvVarNotSetException, IOException,
+            FeStartException, NotInitException, DdlException, InterruptedException {
+        // get DORIS_HOME
+        final String dorisHome = System.getenv("DORIS_HOME");
+        if (Strings.isNullOrEmpty(dorisHome)) {
+            throw new EnvVarNotSetException("env DORIS_HOME is not set");
+        }
+
+        getRandomPort();
+
+        // start fe in "DORIS_HOME/fe/mocked/"
+        MockedFrontend frontend = MockedFrontend.getInstance();
+        Map<String, String> feConfMap = Maps.newHashMap();
+        // set additional fe config
+        feConfMap.put("http_port", String.valueOf(fe_http_port));
+        feConfMap.put("rpc_port", String.valueOf(fe_rpc_port));
+        feConfMap.put("query_port", String.valueOf(fe_query_port));
+        feConfMap.put("edit_log_port", String.valueOf(fe_edit_log_port));
+        feConfMap.put("tablet_create_timeout_second", "10");
+        frontend.init(dorisHome + "/" + runningDir, feConfMap);
+        frontend.start(new String[0]);
+
+        // start be
+        MockedBackend backend = MockedBackendFactory.createBackend("127.0.0.1",
+                be_heartbeat_port, be_thrift_port, be_brpc_port, be_http_port,
+                new DefaultHeartbeatServiceImpl(be_thrift_port, be_http_port, be_brpc_port),
+                new DefaultBeThriftServiceImpl(), new DefaultPBackendServiceImpl());
+        backend.setFeAddress(new TNetworkAddress("127.0.0.1", frontend.getRpcPort()));
+        backend.start();
+
+        // add be
+        List<Pair<String, Integer>> bes = Lists.newArrayList();
+        bes.add(Pair.create(backend.getHost(), backend.getHeartbeatPort()));
+        Catalog.getCurrentSystemInfo().addBackends(bes, false, "default_cluster");
+
+        // sleep to wait first heartbeat
+        Thread.sleep(6000);
+    }
+
+    // generate all port from between 20000 ~ 30000
+    private static void getRandomPort() {
```

Comment:
`getRandomPort`  is duplicate with `AnotherDemoTest`. could move to `UtFrameUtils `

Reply:
Which ports are used is all depends on the caller. Here is just an example to get random ports. And ports set here are all static members of the certain Test class. So the only thing 
that can be put to `UtFrameUtils` is:

```
Random r = new Random(System.currentTimeMillis());
int basePort = 20000 + r.nextInt(9000);
```

It seems unnecessary. 

`AnotherDemoTest` is just a copy of `DemoTest`, so don't worry about this part of duplicate codes.

Reply:
OK